### PR TITLE
Formatting GoDoc special comment blocks

### PR DESCRIPTION
**Pull request by Daniel-M**
I've limited the length of all the special comment block on the `.go` files. I've used an `awk` script to do it, checked all the special comments character count and none of them gave a length higher than 78 characters including the `//` chars.

### CODE REVIEW

File: api.go [hunk 1]
```diff
@@ -1,5 +1,7 @@
-// Copyright 2015-2016 mrd0ll4r and contributors. All rights reserved.
-// Use of this source code is governed by the MIT license, which can be found in
+// Copyright 2015-2016 mrd0ll4r and contributors. All rights
```
Comment (mrd0ll4r) [inline api.go]: The copyright header is not godoc and does not need to be 80 chars wide

File: api.go [hunk 2]
```diff
@@ -36,16 +38,20 @@ func (u *BotUpdate) Update() Update {
 	return u.update
 }
 
-// Error returns != nil, if an error occurred during retrieval of the update
+// Error returns != nil, if an error occurred during retrieval of
```
Comment (mrd0ll4r) [inline api.go]: This was originally 77 chars wide, does not need a line break

**Comment by mrd0ll4r** [comment]
I think something about your script is not working correctly - see the comment I marked that originally had a length of 77 chars.

While you're at it, you could also make sure that all godoc comments are in the proper format (we need correct punctuation):
https://blog.golang.org/godoc-documenting-go-code

**Comment by Daniel-M** [comment]
Restructured comment blocks with an improved script, Now it passes travis tests.

File: api.go [hunk 3]
```diff
@@ -173,9 +179,11 @@ func (api *TelegramBotAPI) GetMe() (*UserResponse, error) {
 	return resp, nil
 }
 
-// GetFile returns a FileResponse containing a Path string needed to download a file.
+// GetFile returns a FileResponse containing a Path string needed to
+// download a file.
 // You will have to construct the download link manually like
-// https://api.telegram.org/file/bot<token>/<file_path>, where <file_path> is taken from the response.
+// https:// api.telegram.org/file/bot<token>/<file_path>, where <file_path>
```
Comment (mrd0ll4r) [inline api.go]: There was a space added to the URI

File: doc.go [hunk 4]
```diff
@@ -2,15 +2,20 @@
...
+// Package tbotapi provides a Go wrapper for the Telegram Messenger Bot
+// API.
 //
-// Note that, if the REST API returns an error, that error will be wrapped in a Go error.
+// Note that, if the REST API returns an error, that error will be wrapped
+// in a Go error.
 //
-// We currently only support long polling (i.e. no webhooks). Feature-wise, everything up to and including the January 20
+// We currently only support long polling (i.e. no webhooks). Feature-wise,
+// everything up to and including the January 20
```
Comment (mrd0ll4r) [inline doc.go]: pull the next line up

File: doc.go [hunk 5]
```diff
@@ -2,15 +2,20 @@
...
 // changes should be implemented.
 //
 // Examples are provided in the examples package, so check that out.
 //
-// The Bot API imposes certain limitations, these are especially interesting for inline query results and files. This
-// library does not keep track of those limitations, so you'll have to perform checks yourself.
+// The Bot API imposes certain limitations, these are especially
+// interesting for inline query results and files. This
```
Comment (mrd0ll4r) [inline doc.go]: pull the next line up

**Comment by Daniel-M** [comment]
What I did,   
- Joined lines you've pointed out.  
- Added punctuation to documentation comments requiring it.  
- Fixed the URIs splitted by script. (The ones you've pointed and others).  
- Reformated non-documentation comments so the format looks more uniform.

File: examples/boilerplate/boilerplate.go [hunk 6]
```diff
@@ -1,6 +1,6 @@
 // Copyright 2015-2016 mrd0ll4r and contributors. All rights reserved.
 // Use of this source code is governed by the MIT license, which can be found in
-// the LICENSE file.
+// The LICENSE file.
```
Comment (mrd0ll4r) [inline examples/boilerplate/boilerplate.go]: This doesn't need to be capitalized.

File: examples/boilerplate/boilerplate.go [hunk 7]
```diff
@@ -15,12 +15,13 @@ import (
...
 
-// BotFunc describes how the bot handles an update
+// BotFunc describes how the bot handles an update.
 type BotFunc func(update tbotapi.Update, api *tbotapi.TelegramBotAPI)
 
 // RunBot runs a bot.
 // THIS IS JUST FOR DEMONSTRATION! NOT TO BE USED IN PRODUCTION!
-// It will block until either something very bad happens or closing is closed.
+// It will block until either something very bad happens or closing is
+// Closed.
```
Comment (mrd0ll4r) [inline examples/boilerplate/boilerplate.go]: This doesn't need to be capitalized either.

File: examples/keyboardMarkup.go [hunk 8]
```diff
@@ -21,29 +21,29 @@ func main() {
...
+				// Ignore messages _about_ channels (bots cannot receive from channels, but will probably get events like channel creations).
 				fmt.Println("Ignoring channel message")
 				return
 			}
 
-			// display the incoming message
-			// msg.Chat implements fmt.Stringer, so it'll display nicely
-			// MessageType implements fmt.Stringer, so it'll display nicely
+			// Display the incoming message.
+			// Msg.Chat implements fmt.Stringer, so it'll display nicely.
```
Comment (mrd0ll4r) [inline examples/keyboardMarkup.go]: Don't capitalize here, `msg` refers to the variable.

File: examples/markup.go [hunk 9]
```diff
@@ -20,18 +20,18 @@ func main() {
...
 				return
 			}
-			// Note: Bots cannot receive from channels, at least no text messages. So we don't have to distinguish anything here
+			// Note: Bots cannot receive from channels, at least no text messages. So we don't have to distinguish anything here.
 
-			// display the incoming message
-			// msg.Chat implements fmt.Stringer, so it'll display nicely
-			// we know it's a text message, so we can safely use the Message.Text pointer
+			// Display the incoming message.
+			// Msg.Chat implements fmt.Stringer, so it'll display nicely.
```
Comment (mrd0ll4r) [inline examples/markup.go]: Same here.

File: examples/photo.go [hunk 10]
```diff
@@ -21,25 +21,25 @@ func main() {
...
 				return
 			}
-			// Note: Bots cannot receive from channels, at least no text messages. So we don't have to distinguish anything here
+			// Note: Bots cannot receive from channels, at least no text messages. So we don't have to distinguish anything here.
 
-			// display the incoming message
-			// msg.Chat implements fmt.Stringer, so it'll display nicely
-			// we know it's a text message, so we can safely use the Message.Text pointer
+			// Display the incoming message.
+			// Msg.Chat implements fmt.Stringer, so it'll display nicely.
```
Comment (mrd0ll4r) [inline examples/photo.go]: Same here.

File: incoming.go [hunk 11]
```diff
@@ -333,15 +336,16 @@ type PhotoSize struct {
 	Height int `json:"height"`
 }
 
-// Sticker represents a sticker
+// Sticker represents a sticke.
```
Comment (mrd0ll4r) [inline incoming.go]: This is missing an r.

File: incoming.go [hunk 12]
```diff
@@ -333,15 +336,16 @@ type PhotoSize struct {
...
 type Sticker struct {
 	FileBase
 	Width     int       `json:"width"`
 	Height    int       `json:"height"`
 	Thumbnail PhotoSize `json:"thumb"`
 }
 
-// UpdateResponse represents the response sent by the API for a GetUpdates request
+// UpdateResponse represents the response sent by the API for a GetUpdate
+// request
```
Comment (mrd0ll4r) [inline incoming.go]: This needs a full stop at the end.

File: incoming.go [hunk 13]
```diff
@@ -433,7 +437,8 @@ func (u User) String() string {
 	return fmt.Sprintf("%d/%s", u.ID, u.FirstName)
 }
 
-// UserProfilePhotosResponse represents the response sent by the API on a GetUserProfilePhotos request
+// UserProfilePhotosResponse represents the response sent by the API on a
+// GetUserProfilePhotos request
```
Comment (mrd0ll4r) [inline incoming.go]: This needs a full stop.

File: outgoing.go [hunk 14]
```diff
@@ -436,15 +449,18 @@ type InlineKeyboardButton struct {
 	SwitchInlineQuery string `json:"switch_inline_query,omitempty"`
 }
 
-// ForceReply represents the values sent by a bot so that clients will be presented with a forced reply, see https://core.telegram.org/bots/api#forcereply
+// ForceReply represents the values sent by a bot so that clients will be
+// presented with a forced reply,
+// see https://core.telegram.org/bots/api#forcereply
```
Comment (mrd0ll4r) [inline outgoing.go]: You can pull some of this up.

File: doc.go [hunk 15]
```diff
@@ -2,15 +2,18 @@
...
-// Package tbotapi provides a Go wrapper for the Telegram Messenger Bot API.
+// Package tbotapi provides a Go wrapper for the Telegram Messenger Bot
+// API.
 //
-// Note that, if the REST API returns an error, that error will be wrapped in a Go error.
+// Note that, if the REST API returns an error, that error will be wrapped
+// in a Go error.
 //
 // We currently only support long polling (i.e. no webhooks). Feature-wise, everything up to and including the January 20
 // changes should be implemented.
```
Comment (mrd0ll4r) [inline doc.go]: reformat these two lines

File: doc.go [hunk 16]
```diff
@@ -2,15 +2,18 @@
...
 //
 // Examples are provided in the examples package, so check that out.
 //
 // The Bot API imposes certain limitations, these are especially interesting for inline query results and files. This
```
Comment (mrd0ll4r) [inline doc.go]: pull up the rest of then sentence, then reformat this line

File: examples/chatAction.go [hunk 17]
```diff
@@ -1,6 +1,6 @@
 // Copyright 2015-2016 mrd0ll4r and contributors. All rights reserved.
 // Use of this source code is governed by the MIT license, which can be found in
-// the LICENSE file.
+// The LICENSE file.
```
Comment (mrd0ll4r) [inline examples/chatAction.go]: don't change the license header.

File: examples/echo.go [hunk 18]
```diff
@@ -1,6 +1,6 @@
 // Copyright 2015-2016 mrd0ll4r and contributors. All rights reserved.
 // Use of this source code is governed by the MIT license, which can be found in
-// the LICENSE file.
+// The LICENSE file.
```
Comment (mrd0ll4r) [inline examples/echo.go]: don't change the license header

**Comment by mrd0ll4r** [comment]
LGTM

**Pull request merged**